### ISSUE TITLE
Feature Request - Expire time for Notifications 

### ISSUE THREAD

**Issue by rwillett** [OneSignal/OneSignal-Cordova-SDK#457]
**Description:**
We are trying to work out if it is possible to clear out existing notifications on IOS and Android when the application is closed or backgrounded.

Our application sends out personalised traffic updates to users, a traffic update will have a life of it's own and will change over time, e.g. 

1. the initial update could be "Collision, lane blocked" (this is a simple summary)
2. five mins later, police could be on the soon and they have closed both lanes of a road. This means the severity of the incident has gone up.
3. Five mins later, it could be that the traffic jam has become severe as the road is closed affecting surrounding areas.
4. 20 mins later, the police could clear the accident, but the traffic jams will take time to clear
5. 20 mins later there could still be residual traffic jams.

We would send push notification updates every 5 mins or so. This means that older updates become stale and no longer relevant. Traffic news becomes old news very quickly :)

Currently our system sends the push notification updates and they simply stack up. A new update is far more useful than an older update, even if the older update is only five mins old. 

What we would like to do is expire an old update when a new update appears, so any new update clears out all older updates (of any incident).

Every incident has a unique for the incident which never changes over the lifetime of the incident.

We have looked through the SDK and cannot see any method of removing old notifications, we know that we can group them, but that's not an ideal solution. As far as we can see, expiring notifications only works for when a user logs on again.

We realise that this may not be possible but are asking in case there is a special technique we are not aware of, or some clever person has done this through setting the badges in a special order, or whatever.

**Environment**
onesignal-cordova-plugin 2.4.3 "OneSignal Push Notifications"

Not sure if any other plugin helps hetre :)

**Steps to Reproduce Issue:**

This is not a big, possibly a feature request.

**Anything else:**
We have looked through all the closed and open issues and can't see anybody else doing what we want to do.

**Comment by jkasten2** [comment]
@rwillett Thanks for all the details and looking through the previous tickets for anythings similar.

There isn't an option to expire notifications built into Android or iOS or this SDK but there are a few others things you can do to make sure users are not seeing old traffic information.
1. Set the notification time to live to 5 minutes -  Set `ttl` to `300`.
2. Overwrite notifications for the same traffic event -  Use the same notification `collapse_id`.
See the OneSignal REST API docs for more details on these parameters.
https://documentation.onesignal.com/reference#create-notification

As far as auto clearing old notifications after some time while the app is in the background / close it will take some native code to do this. For Android something could be done by setting up and scheduling a Broadcast. https://stackoverflow.com/a/23874764/1244574

We will consider this use case when we are adding new SDK features. Thanks.

**Comment by rwillett** [comment]
Josh,

Thanks for the helpful reply. We already set the TTL to 300 to try and clear notifications ASAP.
We did not see the ```collapse_id``` field before. This is our fault for not reading the docs properly. We'll implement that and that will help. 

The only downside is that the last notification in the sequence is always left behind as there is nothing to replace it. However this does move us on. We haven't got the ideal fix but we can reduce the number of old notifications. 

I would suggest we leave this open and tag it as an Enhancement.

Thanks

Rob
